dash/dash.py

```diff
@@ -378,15 +378,27 @@ def _relative_url_path(relative_package_path='', namespace=''):
 
         srcs = []
         for resource in resources:
+            is_dynamic_resource = (
+                'dynamic' in resource and
+                resource['dynamic'] is True
+            )
+
             if 'relative_package_path' in resource:
                 if isinstance(resource['relative_package_path'], str):
-                    srcs.append(_relative_url_path(**resource))
+                    path = _relative_url_path(
+                        resource['relative_package_path'],
+                        resource['namespace']
+                    )
+                    if not is_dynamic_resource:
```

Comment:
We need to figure out what the relative path is because that's also where we register the file -- but only add it to srcs (will be used below for ```<script />``` entries if it's *not* dynamic

Reply:
Can we instead just move the `self.registered_paths[namespace].add(relative_package_path)` out? I'm not sure why that logic is there in the first place, but now it is confusing to create an un-used local variable `path` in the case that `not is_dynamic_resource` only trigger the side effect of `_relative_url_path`. 

I think that would make this portion easier to understand, e.g.
```
self.registered_paths[namespace].add(relative_package_path)
if not is_dynamic_resource:
    srcs.append(_relative_url_path(
        resource['relative_package_path'],
        resource['namespace']
    ))
```